doc/man3/SSL_in_init.pod

```diff
@@ -0,0 +1,110 @@
+=pod
+
+=head1 NAME
+
+SSL_in_before,
+SSL_in_init,
+SSL_is_init_finished,
+SSL_in_connect_init,
+SSL_in_accept_init,
+SSL_get_state
+- retrieve information about the handshake state machine
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_in_init(SSL *s);
+ int SSL_in_before(SSL *s);
+ int SSL_is_init_finished(SSL *s);
+
+ int SSL_in_connect_init(SSL *s);
+ int SSL_in_accept_init(SSL *s);
+
+ OSSL_HANDSHAKE_STATE SSL_get_state(const SSL *ssl);
+
+=head1 DESCRIPTION
+
+SSL_in_init() returns 1 if the SSL/TLS state machine is currently processing or
+awaiting handshake messages, or 0 otherwise.
+
+SSL_in_before() returns 1 if no SSL/TLS handshake has yet been initiated, or 0
+otherwise.
```

Comment:
Well, I don't really see the "bend of the truth". Basically there are two different things - the "in before" state in the state machine, and what "SSL_in_before()" returns. The comment is just (trying) to explain the difference between the two things. The function SSL_in_before() returns 1 when no SSL/TLS handshake has yet been initiated (as the documentation states). However the state machine may still be in the "in before" state for a short while after SSL_in_before() no longer returns 1, i.e. the application has asked OpenSSL to start the handshake process, but as yet no messages have actually been processed. In other words I think the docs I have written are consistent with the comment.